### DISCUSSION TITLE
ESSIF, W3C DID & W3C Verifiable Credentials Compliance

### DISCUSSION THREAD

**Discussion by Alex-DAW** [nmshd/feedback#20] [Q&A]
Hi, i would like to know if Enmesh works with any of the following standards defined by the European Union and W3C for the issuance of verifiable credentials as well as interoperability with other wallets to share credentials.

- [Decentralized Identifiers](https://www.w3.org/TR/did-core/)
- [Verifiable Credentials Data Model](https://www.w3.org/TR/vc-data-model/)
- [European Self Sovereign Identity Framework]

It's possible to issue a verifiable credential within the Enmesh ecosystem?

It's also possible to a user of Enmesh app to receive a verifiable credential from any issuer and store inside the wallet?

Thank you!

**Comment by stnmtz** [comment]
Hi Alex,

thanks for reaching out to us.

> It's also possible to a user of Enmesh app to receive a verifiable credential from any issuer and store inside the wallet?

Although enmeshed doesn't fully support Verifiable Credentials (VC) and Verifiable Presentations (VP) right now, you can communicate them by enmeshed as unstructured content and we are certainly looking further into the topic.

> It's possible to issue a verifiable credential within the Enmesh ecosystem?

Not as of now, but we are working on it. From a technology standpoint, we have working proof of concepts to create did:key's and sign VCs with enmeshed Identities. Still the question would be, if you trust the issuer and how to know the issuer. 

Understanding (rendering and validating) certain VC data models or issuers is the hard part, as it is quite open and we do not want to open up the enmeshed App (or even the Connector) to have access to the whole Internet. Thus, the JSON-LD datamodels of the VCs and the trusted issuers should be known upfront and stored into the respective components to enhance privacy and security. Additionally, there doesn't seem to be any finalized VC data model for any domain right now.

Although VCs are no first-class citizen in the enmeshed space so far (i.e. having their own attributes), they could still be communicated between the Identities as unstructured data. Same with signed and unsigned PDF files, images or any other blob.

In short, yes you can communicate VCs over enmeshed, no you cannot issue or validate VCs as of now, but we are working on it. VPs is a tougher nut and we need to tackle this after general support of VCs.

What exactly do you have in mind with VCs and how would you tackle them with enmeshed? We would love to hear your thoughts about it.

Cheers

**Reply by Alex-DAW** [reply]
Hi Thomas, 

thank you for your answer, i really appreciate it.

We're working on a EUDI (European Union Digital Identity) wallet project aimed to issue and store education digital certificates following the European and W3C standards to achieve interoperability between wallets. 

We're aware that data models or schemas are not properly defined yet, but we believe that in a near future it'll exist some generic schemas to describe the main data of any credential type. Moreover all this standards make us think that in the future they will be created ESSIF compliant wallets to handle many types of scenarios besides the digital identity of the user.

Our main goal is to issue education VC to the users once they complete a course and allow the users to store the VC on a ESSIF compliant wallet. So is interesting for us to know if Enmeshed will be ESSIF compliant in the future and what will be your estimate timeline.

Thank you!
